### DISCUSSION TITLE
Vivado simulation waveform

### DISCUSSION THREAD

**Discussion by jerome2406** [NUS-CG3207/labs#7] [Lab 1] [answered]
For the simulation waveform, how do we ensure that "all signal names should be captured"?
I tried simulating the top level module but the intermediate signals don't show up. Is there a way to make them show up or will these signals suffice?
![image](https://github.com/user-attachments/assets/daee41d8-7bf4-42ae-b39a-d7c17df6fe09)

**Comment by NeilBaner** [comment, marked as answer]
It'll be good if you can include your counter as well. Try peeking around to see if you can find the way to add that waveform to the simulation. It'll be useful to be able to do this in future labs too :)

**Reply by jerome2406** [reply]
https://stackoverflow.com/questions/38018845/how-to-display-module-variables-in-a-waveform-window-in-vivado#:~:text=1%20Answer&text=If%20you%20click%20the%20UUT,only%20see%20the%20testbench%20variables.

if anyone needs also.
